tests/integration_tests/modules/test_lxd.py

```diff
@@ -199,12 +200,28 @@ def validate_preseed_storage_pools(client, preseed_cfg):
         assert storage_pool["driver"] == src_storage["driver"]
 
 
-def validate_preseed_projects(client, preseed_cfg):
+def validate_preseed_projects(client: IntegrationInstance, preseed_cfg):
+    # Support for projects by lxd init --preseed was added in lxd 4.12
```

Comment:
Being this true, I wonder if `cloud-init` should issue some kind of warning to better inform users about this fact.

So if the user data contains a lxd preseed with project config and lxd < 4.12, then the project config is ignored by lxd and cloud-init does not notify the user.

Reply:
I'd lean toward filing a bug agains LXD for this. While I expect this is going to be outside of support for features LXD will grow for LXD < 4.12, it'd be good to track this shortfall in preseed validation in LXD. 

We walk a fine line in cloud-init when we "support" opaque keys that pass through direct to external software/services. I'd prefer if cloud-init doesn't grow this static knowledge about whether or not 'projects' or any other top-level config key is supported by a particular version of LXD preseed. This support of "projects" config on a the LXD 4.0-stream can change over time and cloud-init hard-coding expectations and warnings about those cases seems a bit out of scope.

Reply:
Agreed, I created an LXD issue representing this: https://github.com/lxc/lxd/issues/11106